mautic/mautic
# Added new GROUP_CONCAT function to reports

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/6198
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Based on this issue https://github.com/mautic/mautic/issues/6198 
Reports with join tables and multiple results give multiple rows for on contact.
This PR added GROUP_CONCAT function allow add column with coma separated results in one row (for example companies to one line to concat)
More https://dev.mysql.com/doc/refman/8.0/en/group-by-functions.html#function_group-concat

Before

![image](https://user-images.githubusercontent.com/462477/42289638-0a94a9c8-7fc1-11e8-93b8-0ca4a40a85f2.png)

After
![image](https://user-images.githubusercontent.com/462477/42289662-38575b6c-7fc1-11e8-93cc-ece3eef9c149.png)

#### Steps to test this PR:
1. Create contact with two or more assign companies and submit form with his ID
2.  Create form submisision report
3.  Setup like me on screenshot

![image](https://user-images.githubusercontent.com/462477/42289685-5695b20e-7fc1-11e8-8dac-6b31308db633.png)

4. Expect this column (separated values with comma)
![image](https://user-images.githubusercontent.com/462477/42289700-6ac6f6d4-7fc1-11e8-92a7-4120e2a7ab6a.png)
